feat: multiple versions for the vault extension

jfroche opened the pull request:
Build multiple versions of the vault extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

jfroche committed:
feat: multiple versions for the vault extension

Build multiple versions of the vault extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

hunleyd reviewed (changes requested):
rebase to fix conflicts?

jfroche committed:
fix: nixos tests for pg_repack, pgroonga, and plpgsql_check to use the supabase_admin user

We switched to using the supabase_admin user for our Postgres tests instead of the default postgres user. 
This change updates the authentication methods and user management in the NixOS test configurations for the pg_repack, pgroonga, and plpgsql_check extensions.

jfroche committed:
fix: nixos test for plv8 extension to use the supabase_admin user

We switched to using the supabase_admin user for our Postgres tests instead of the default postgres user.
This change updates the authentication methods and user management in the NixOS test configurations for the plv8 extension.

samrose committed:
chore: bump one more
